internal/blueprint/internal/scheme/transform.go

```diff
@@ -0,0 +1,315 @@
+package scheme
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"github.com/icza/dyno"
+)
+
+func toInternal(obj any) any {
+	if getString(obj, "apiVersion") != "g2a-cli/v2.0" {
+		obj = toV2(obj)
+	}
+
+	switch kind := getString(obj, "kind"); kind {
+	case "Project":
+		return toInternalProject(obj)
+	case "Service":
+		return toInternalService(obj)
+	case "Environment":
+		return toInternalEnvironment(obj)
+	case "Builder", "Deployer", "Pusher", "Tagger":
+		return toInternalExecutor(obj)
+	default:
+		panic(fmt.Errorf("unsupported kind: %s", kind))
+	}
+}
+
+func toInternalProject(obj any) any {
+	files := []any{}
+
+	for _, f := range getSlice(obj, "files") {
```

Comment:
This whole function should be documented in details, otherwise this logic seems like something you make up as you go. Those are mappings, right? So those mappings need to be described as an algorithm - A defined by some structure can be mapped into B defined into some other structure.

This blocks review for me. As I'm not sure what is the sources and what is the target of a transformation.